### PR TITLE
[FIX] stock: validate an immediate transfer without a backorder

### PR DESCRIPTION
**Pull request by DjamelTouati**
Steps to reproduce the bug:
- Create a Storable product “C1” tracked by Lot:
    - Update the Qty to 10

- Create a Storable product “P1”:
    - Add a BoM:
        - add 1 unit of  “C1” as component

- Enable 3 steps for the manufacturing operation in warehouse settings

- Create a Mo to produce 1 unit of “P1”:
    - Confirm the MO
    - Click on related transfer:
        - Select the “Pick component”
        - check that the qty is reserved correctly
        - Try to validate it without setting Qty done
        - The immediate transfer is triggered, validate it

Problem:
The backorder's wizard triggers when it shouldn't

To know if we should create a backorder, we check if the qty reserved is equal to the qty done, and as the qty done was not set correctly during the validation of the immediate transfer, the two quantities are not identical, therefore the widget is trigger:

https://github.com/odoo/odoo/blob/15.0/addons/stock/models/stock_picking.py#L1128-L1132

opw-3240264

### CODE REVIEW

**Comment by Billolha** [comment]
Hi this happens also on 16.0

File: addons/stock/tests/test_move.py [hunk 1]
```diff
@@ -4602,6 +4602,36 @@ def test_immediate_validate_9_tracked_move_with_0_qty_done(self):
         immediate_wizard_form.process()
         self.assertEqual(receipt.state, 'done')
 
+    def test_immediate_validate_10_tracked_move_without_backorder(self):
+        """
+            Create a picking for a tracked product, validate it as an
+            immediate transfer, and ensure that the backorder wizard is
+            not triggered when the qty is reserved.
+        """
+        picking_type_receipt = self.env.ref('stock.picking_type_internal')
```
Comment (svs-odoo) [inline addons/stock/tests/test_move.py:4611]:
```suggestion
        picking_type_internal = self.env.ref('stock.picking_type_internal')
```
Same for the picking's variable=
```python
        internal_transfer = picking_form.save()
```

File: addons/stock/tests/test_move.py [hunk 2]
```diff
@@ -4602,6 +4602,36 @@ def test_immediate_validate_9_tracked_move_with_0_qty_done(self):
...
+            'company_id': self.env.company.id,
+        })
+        self.env['stock.quant']._update_available_quantity(self.product_lot, self.stock_location, 10, lot_id=lot)
+        picking_form = Form(self.env['stock.picking'])
+        picking_form.picking_type_id = picking_type_internal
+        with picking_form.move_ids_without_package.new() as move:
+            move.product_id = self.product_lot
+            move.product_uom_qty = 4
+        receipt = picking_form.save()
+        receipt.action_confirm()
```
Comment (svs-odoo) [inline addons/stock/tests/test_move.py:4626]: > Same for the picking's variable=
> ```python
>        internal_transfer = picking_form.save()
>```

Comment (DjamelTouati) [reply addons/stock/tests/test_move.py]: Nice catch, thanks for your review !

**Comment by svs-odoo** [comment]
robodoo r+

**Pull request closed without merging**